### PR TITLE
Additional test settings for non-default hyperparameters

### PR DESCRIPTION
**Pull request by sbharadwajj**
Added test settings (New format), for `extensions/firstorder`

### CODE REVIEW

File: test/core/derivatives/convolution_settings.py [hunk 1]
```diff
@@ -276,3 +276,152 @@
         "id_prefix": "non-default-conv",
     },
 ]
+
+CONVOLUTION_GROUPS_SETTINGS = [
```
Comment (f-dangel) [inline test/core/derivatives/convolution_settings.py:280]: I think you are overwriting the settings here, should it be `+=`?

File: backpack/core/derivatives/conv_transposend.py [hunk 2]
```diff
@@ -52,6 +52,9 @@ def _bias_jac_mat_prod(self, module, g_inp, g_out, mat):
         return jac_mat.expand(*expand_shape)
 
     def _weight_jac_mat_prod(self, module, g_inp, g_out, mat):
+        if module.groups != 1:
+            raise NotImplementedError
```
Comment (f-dangel) [inline backpack/core/derivatives/conv_transposend.py:56]: Can you add a message to the exeception? Something like "groups ≠ 1 is not supported"

File: test/core/derivatives/convolution_settings.py (hunk 1 above)
Comment (f-dangel) [reply test/core/derivatives/convolution_settings.py:280]: Maybe I also missunderstood something here. Where is `CONVOLUTION_GROUPS_SETTING` being used in the tests?

File: test/extensions/firstorder/firstorder_settings.py [hunk 3]
```diff
@@ -100,17 +113,41 @@
...
+            torch.nn.Linear(15, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="sum"),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
     {
         "input_fn": lambda: torch.rand(3, 3, 7, 7),
         "module_fn": lambda: torch.nn.Sequential(
-            torch.nn.Conv2d(3, 2, 2),
+            torch.nn.Conv2d(3, 2, 2, bias=False),
```
Comment (f-dangel) [inline test/extensions/firstorder/firstorder_settings.py:130]: What's the reason the bias had to go?

File: test/extensions/firstorder/firstorder_settings.py [hunk 4]
```diff
@@ -122,17 +159,41 @@
...
+            torch.nn.Linear(75, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
     {
         "input_fn": lambda: torch.rand(3, 3, 2, 7, 7),
         "module_fn": lambda: torch.nn.Sequential(
-            torch.nn.Conv3d(3, 2, 2),
+            torch.nn.Conv3d(3, 2, 2, bias=False),
```
Comment (f-dangel) [inline test/extensions/firstorder/firstorder_settings.py:176]: What's the reason the bias had to go?

Comment (sbharadwajj) [reply test/extensions/firstorder/firstorder_settings.py:176]: Bias was added.

File: test/core/derivatives/convolution_settings.py (hunk 1 above)
Comment (sbharadwajj) [reply test/core/derivatives/convolution_settings.py:280]: This is another confusion I had. Did you want this to be like `LOSS_FAIL_SETTINGS`?

Comment (sbharadwajj) [reply test/core/derivatives/convolution_settings.py:280]: Its not used anywhere as the settings here fail for `weights`. But maybe once I raise `NotImplementedError` correctly we wont need this

File: test/core/derivatives/derivatives_test.py [hunk 5]
```diff
@@ -200,6 +204,21 @@ def test_sqrt_hessian_squared_equals_hessian(problem):
...
+def test_weight_jac_mat_prod_should_fail(problem):
+    with pytest.raises(ValueError):
+        test_weight_jac_mat_prod(problem)
+
+
+@pytest.mark.parametrize(
+    "sum_batch", [True, False], ids=["sum_batch=True", "sum_batch=False"]
+)
+@pytest.mark.parametrize("problem", CONVOLUTION_FAIL_PROBLEMS, ids=CONVOLUTION_FAIL_IDS)
+def test_weight_jac_t__mat_prod_should_fail(problem, sum_batch):
```
Comment (f-dangel) [inline test/core/derivatives/derivatives_test.py:217]: `jac_t__` -> `jac_t_`

File: backpack/core/derivatives/conv_transposend.py [hunk 6]
```diff
@@ -52,47 +52,53 @@ def _bias_jac_mat_prod(self, module, g_inp, g_out, mat):
...
+            )
+
+            dims_kern = "xyz"[: self.conv_dims]
+            dims_data = "abc"[: self.conv_dims]
+            einstr = "nig{0}{1},vigo{0}->vngo{1}".format(dims_kern, dims_data)
+            jac_mat = einsum(einstr, u, mat_reshape)
+
+            return self.reshape_like_output(jac_mat, module)
+        else:
+            raise NotImplementedError("Groups greater than 1 are not supported yet")
```
Comment (f-dangel) [inline backpack/core/derivatives/conv_transposend.py:74]: I would revert the order: raise the exception if `groups==1`. Then you avoid indenting the actual code.

File: backpack/core/derivatives/conv_transposend.py [hunk 7]
```diff
@@ -52,47 +52,53 @@ def _bias_jac_mat_prod(self, module, g_inp, g_out, mat):
...
-            (V, *module.weight.shape) if sum_batch else (V, N, *module.weight.shape)
-        )
+            final_shape = (
+                (V, *module.weight.shape) if sum_batch else (V, N, *module.weight.shape)
+            )
 
-        return einsum(equation, u, mat_reshape).reshape(final_shape)
+            return einsum(equation, u, mat_reshape).reshape(final_shape)
+        else:
+            raise NotImplementedError("Groups greater than 1 are not supported yet")
```
Comment (f-dangel) [inline backpack/core/derivatives/conv_transposend.py:101]: same as above: I would revert the order

File: test/core/derivatives/derivatives_test.py [hunk 8]
```diff
@@ -200,6 +204,21 @@ def test_sqrt_hessian_squared_equals_hessian(problem):
     problem.tear_down()
 
 
+@pytest.mark.parametrize("problem", CONVOLUTION_FAIL_PROBLEMS, ids=CONVOLUTION_FAIL_IDS)
+def test_weight_jac_mat_prod_should_fail(problem):
+    with pytest.raises(ValueError):
```
Comment (f-dangel) [inline test/core/derivatives/derivatives_test.py:209]: this should be waiting for the `NotImplementedError` to be raised

File: test/core/derivatives/derivatives_test.py [hunk 9]
```diff
@@ -200,6 +204,21 @@ def test_sqrt_hessian_squared_equals_hessian(problem):
...
+        test_weight_jac_mat_prod(problem)
+
+
+@pytest.mark.parametrize(
+    "sum_batch", [True, False], ids=["sum_batch=True", "sum_batch=False"]
+)
+@pytest.mark.parametrize("problem", CONVOLUTION_FAIL_PROBLEMS, ids=CONVOLUTION_FAIL_IDS)
+def test_weight_jac_t__mat_prod_should_fail(problem, sum_batch):
+    with pytest.raises(ValueError):
```
Comment (f-dangel) [inline test/core/derivatives/derivatives_test.py:218]: same as above: `NotImplementedError`

File: test/extensions/firstorder/firstorder_settings.py [hunk 10]
```diff
@@ -81,14 +81,27 @@
     {
         "input_fn": lambda: torch.rand(3, 3, 7),
         "module_fn": lambda: torch.nn.Sequential(
-            torch.nn.Conv1d(3, 2, 2),
+            torch.nn.Conv1d(3, 2, 2, bias=False),
```
Comment (f-dangel) [inline test/extensions/firstorder/firstorder_settings.py:84]: Instead of modifying an existing test, could you add a new one with `bias=False`?

**Pull request merged**